fixed definition of the new Leadteck USB stick:
- Leadtek USB DVB-T stick [ID 0413:6a03]
(thanks to djnice);

added definition of 3 new Terratec USB sticks:
- TERRATEC Cinergy T Stick RC (Rev.3) [ID 0ccd:00d3]
- TERRATEC Cinergy T Stick BLACK (Rev.2) [ID 0ccd:00d4]
- TERRATEC NOXON DAB Stick (Rev.2) [ID 0ccd:00e0] 
(thanks to CyBoardR);

From: Gianluca Gennari <gennarone@gmail.com>
---
 drivers/media/dvb/dvb-usb/rtl2832u.c |   41 ++++++++++++++++++++++++++++-----
 drivers/media/dvb/dvb-usb/rtl2832u.h |    3 ++
 2 files changed, 37 insertions(+), 7 deletions(-)

diff --git a/drivers/media/dvb/dvb-usb/rtl2832u.c b/drivers/media/dvb/dvb-usb/rtl2832u.c
index c022bf8..fadb179 100644
--- a/drivers/media/dvb/dvb-usb/rtl2832u.c
+++ b/drivers/media/dvb/dvb-usb/rtl2832u.c
@@ -826,7 +826,6 @@ static struct usb_device_id rtl2832u_usb_table [] = {
 
 	{ USB_DEVICE(USB_VID_LEADTEK, USB_PID_LEADTEK_WARM_1)},		// 60			
 	{ USB_DEVICE(USB_VID_LEADTEK, USB_PID_LEADTEK_WARM_2)},		// 61
-	{ USB_DEVICE(USB_VID_LEADTEK, USB_PID_LEADTEK_WARM_3)},
 
 	{ USB_DEVICE(USB_VID_YUAN, USB_PID_YUAN_WARM)},			//62
         { USB_DEVICE(USB_VID_YUAN, USB_PID_YUAN_WARM80)},		//63
@@ -843,8 +842,14 @@ static struct usb_device_id rtl2832u_usb_table [] = {
 	{ USB_DEVICE(USB_VID_COMPRO, USB_PID_COMPRO_WARM_9530)},	// 73  71																						//------rtl2832u_6th_properties(6)
 	{ USB_DEVICE(USB_VID_COMPRO,  USB_PID_COMPRO_WARM_9520)},	// 74
 	
-	{ USB_DEVICE(USB_VID_GOLDENBRIDGE, USB_PID_GOLDENBRIDGE_WARM)},	//75	
-													
+	{ USB_DEVICE(USB_VID_GOLDENBRIDGE, USB_PID_GOLDENBRIDGE_WARM)},	//75
+
+	{ USB_DEVICE(USB_VID_LEADTEK, USB_PID_LEADTEK_WARM_3)},		// 76
+
+	{ USB_DEVICE(USB_VID_TERRATEC, USB_PID_TERRATEC_00D3)},		// 77
+	{ USB_DEVICE(USB_VID_TERRATEC, USB_PID_TERRATEC_00D4)},		// 78
+	{ USB_DEVICE(USB_VID_TERRATEC, USB_PID_TERRATEC_00E0)},		// 79
+
 	{ 0 },
 };
 
@@ -1207,7 +1212,7 @@ static struct dvb_usb_device_properties rtl2832u_4th_properties = {
 		.rc_interval  = RT_RC_POLLING_INTERVAL_TIME_MS,		
 	},
 	
-	.num_device_descs = 9,
+	.num_device_descs = 12,
 	.devices = {
 		{ .name = "DK DONGLE",
 		  .cold_ids = { NULL, NULL },
@@ -1254,8 +1259,25 @@ static struct dvb_usb_device_properties rtl2832u_4th_properties = {
 		  .cold_ids = { NULL, NULL },
 		  .warm_ids = { &rtl2832u_usb_table[35], NULL },
 		},				
-		
-		
+
+		{
+		  .name = "Terratec Cinergy T Stick RC (Rev.3)",
+		  .cold_ids = { NULL, NULL },
+		  .warm_ids = { &rtl2832u_usb_table[77], NULL },
+		},
+
+		{
+		  .name = "Terratec Cinergy T Stick BLACK (Rev.2)",
+		  .cold_ids = { NULL, NULL },
+		  .warm_ids = { &rtl2832u_usb_table[78], NULL },
+		},
+
+		{
+		  .name = "Terratec Noxon DAB Stick (Rev.2)",
+		  .cold_ids = { NULL, NULL },
+		  .warm_ids = { &rtl2832u_usb_table[79], NULL },
+		},
+
 	}
 };
 
@@ -1527,7 +1549,7 @@ static struct dvb_usb_device_properties rtl2832u_7th_properties = {
 		.rc_interval  = RT_RC_POLLING_INTERVAL_TIME_MS,		
 	},
 	
-	.num_device_descs = 9,
+	.num_device_descs = 10,
 	.devices = {
 		{ .name = "DVB-T TV Stick",
 		  .cold_ids = { NULL, NULL },
@@ -1568,6 +1590,11 @@ static struct dvb_usb_device_properties rtl2832u_7th_properties = {
 		  .cold_ids = { NULL, NULL },
 		  .warm_ids = { &rtl2832u_usb_table[62], NULL },
 		},
+		{
+		  .name ="Leadtek USB DVB-T Device",
+		  .cold_ids = { NULL, NULL },
+		  .warm_ids = { &rtl2832u_usb_table[76], NULL },
+		},
 
 		{ NULL },				
 	}
diff --git a/drivers/media/dvb/dvb-usb/rtl2832u.h b/drivers/media/dvb/dvb-usb/rtl2832u.h
index d4823a1..86e476c 100644
--- a/drivers/media/dvb/dvb-usb/rtl2832u.h
+++ b/drivers/media/dvb/dvb-usb/rtl2832u.h
@@ -48,6 +48,9 @@
 #endif
 #define	USB_PID_TERRATEC_00A9					0x00A9
 #define	USB_PID_TERRATEC_00B3					0x00B3
+#define	USB_PID_TERRATEC_00D3					0x00D3
+#define	USB_PID_TERRATEC_00D4					0x00D4
+#define	USB_PID_TERRATEC_00E0					0x00E0
 
 #ifndef USB_VID_AZUREWAVE_2
 	#define	USB_VID_AZUREWAVE_2					0x13D3
-- 
1.7.0.4

